docs/guides/using-rewrites.md

```diff
@@ -0,0 +1,188 @@
+# Rewriting paths
+This guide demonstrates host and path rewrites using Ingress and Service configuration.
+
+## Installation
+
+Please follow the [deployment](../deployment) documentation to install
+Kong Ingress Controller on your Kubernetes cluster.
+
+## Testing Connectivity to Kong
+
+This guide assumes that the `PROXY_IP` environment variable is
+set to contain the IP address or URL pointing to Kong.
+Please follow one of the
+[deployment guides](../deployment) to configure this environment variable.
+
+If everything is setup correctly, making a request to Kong should return
+HTTP 404 Not Found.
+
+```bash
+$ curl -i $PROXY_IP
+HTTP/1.1 404 Not Found
+Date: Fri, 21 Jun 2019 17:01:07 GMT
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+Content-Length: 48
+Server: kong/1.2.1
+
+{"message":"no Route matched with those values"}
+```
+
+This is expected as Kong does not yet know how to proxy the request.
+
+## Create a test Deployment
+
+To test our requests, we create an echo server Deployment, which responds to
+HTTP requests with a summary of the request contents:
+
+```bash
+$ kubectl create namespace echo
```

Comment:
This goes against all other guides, which are in the default namespace.
What is the motivation behind adding this and diverging from the norm?

Reply:
I'd argue that the norm is a bit debatable--creating example configuration in a dedicated namespace is good because you can follow the example verbatim and clean up quickly after by just deleting the namespace.

There's no requirement to use any given namespace in actual production configuration, but I'd expect users to infer that when adapting the guide to their own config. Maybe not, so I'll add a clarifying note instead.

Reply:
> creating example configuration in a dedicated namespace is good because you can follow the example verbatim and clean up quickly after by just deleting the namespace.

Yes, that sounds good and I suspected that this was the motivation for this. The only concern here is that this is inconsistent with other guides we have but that's okay. We need to start a positive change from _somewhere_ first.

> Maybe not, so I'll add a clarifying note instead.

We expect users to figure those basic details out by themselves. That's not a problem we want to solve. We can provide them a pointer to learn those basics, but that's the most I think.